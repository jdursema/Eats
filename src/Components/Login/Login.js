import React, { Component } from 'react';

class Login extends Component {
  constructor(){
    super()
    this.state ={
      username: '',
      password: ''
    }
  }

  handleStateChange = event => {
    this.setState({[event.target.name] : event.target.value})
  }

  handleSubmit(e) {
    e.preventDefault();
    const userRef = firebase.database().ref('user');
    const user = {
      favorites: [],
      userName: this.state.userName,
    }
  }


  render() {
    return (
      <form>
        <label for='username-input'>Username: </label>
        <input type='text' className='username-input' onChange={this.handleStateChange} name='username'/>
        <label for='password-input'>Password: </label>
        <input type='password' className='password-input' onChange={this.handleStateChange} name='password' value={this.state.password}/>
        <button>Create Users</button>
      </form>
    )
  }
}

export default  Login;