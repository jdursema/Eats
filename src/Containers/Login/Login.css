.login-form {
  margin: 2% auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  border: solid black 3px;
  width: 40%;
  padding: 5%;
  align-content: center;
  align-self: center;
  font-family: 'Raleway', sans-serif;
  color: #0F2130; }

.login-form label {
  border-bottom: lightgrey solid 1px;
  padding: 2%; }

.login-form .password-input {
  height: 18px;
  width: 200px; }

.login-form .email-input {
  height: 18px;
  width: 200px; }

.login-form button {
  background-color: #0F2130;
  color: white;
  border: solid 1px #0F2130;
  font-size: 16px;
  margin: 2%; }
