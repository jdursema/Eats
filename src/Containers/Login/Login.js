import React, { Component } from 'react';
import { checkUser } from '../../Actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './Login.css';


export class Login extends Component {
  constructor(){
    super();
    this.state ={
      email: '',
      password: ''
    };
  }

  handleStateChange = event => {
    this.setState({[event.target.name] : event.target.value});
  }


  render() {
    return (
      <form className = 'login-form' >
        <label htmlFor='email-input'>
          Email: 
        </label>
        <input 
          type='text' 
          className='email-input' 
          onChange={this.handleStateChange}
          name='email'/>
        <label htmlFor='password-input'>
          Password: 
        </label>
        <input 
          type='password' 
          className='password-input' 
          onChange={this.handleStateChange}
          name='password' 
          value={this.state.password}/>
        <button onClick={ (event)=>{
          event.preventDefault();
          this.props.handleLogin(this.state.email, this.state.password);
          if (this.props.user.email){
            this.props.history.push('/suggestions'); 
}
        }}>
            Login
        </button>
        <p className='error'>{this.props.error}</p>
      </form>

    );
  }
}

const mapStateToProps = state => ({
  error: state.error,
  user: state.user
});

const mapDispatchToProps = dispatch => ({
  handleLogin: (email, password) => dispatch(checkUser(email, password))
});
    

export default connect(mapStateToProps, mapDispatchToProps)(Login);


Login.propTypes = {
  handleLogin: PropTypes.func,
  error: PropTypes.string
};