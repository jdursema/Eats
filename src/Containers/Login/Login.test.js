import { Login } from './Login';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe ('Login', () => {
  it('should match the snapshot', () => {
    const renderedLogin = shallow(<Login />);

    expect(renderedLogin).toMatchSnapshot();
  });

  it('should call handleSignUp when the button is clicked', () => {
    const mockFunc = jest.fn()
    
    const renderedLogin = shallow(<Login handleLogin = {mockFunc}/>);

    const button = renderedLogin.find('button');

    expect(mockFunc.mock.calls.length).toEqual(0);

    button.simulate('click');

    expect(mockFunc.mock.calls.length).toEqual(1);
  });
});