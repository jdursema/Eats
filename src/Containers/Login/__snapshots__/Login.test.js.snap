// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="email-input"
>
          Email:
</label>,
        <input
          className="email-input"
          name="email"
          onChange={[Function]}
          type="text"
/>,
        <label
          htmlFor="password-input"
>
          Password:
</label>,
        <input
          className="password-input"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
/>,
        <button
          onClick={[Function]}
>
          Login
</button>,
        <p />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Email:",
          "htmlFor": "email-input",
        },
        "ref": null,
        "rendered": "Email:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "email-input",
          "name": "email",
          "onChange": [Function],
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Password:",
          "htmlFor": "password-input",
        },
        "ref": null,
        "rendered": "Password:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "password-input",
          "name": "password",
          "onChange": [Function],
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Login",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="email-input"
>
            Email:
</label>,
          <input
            className="email-input"
            name="email"
            onChange={[Function]}
            type="text"
/>,
          <label
            htmlFor="password-input"
>
            Password:
</label>,
          <input
            className="password-input"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
/>,
          <button
            onClick={[Function]}
>
            Login
</button>,
          <p />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Email:",
            "htmlFor": "email-input",
          },
          "ref": null,
          "rendered": "Email:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "email-input",
            "name": "email",
            "onChange": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Password:",
            "htmlFor": "password-input",
          },
          "ref": null,
          "rendered": "Password:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password-input",
            "name": "password",
            "onChange": [Function],
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
